### PR TITLE
Run cni release test for deveks

### PR DESCRIPTION
**Pull request by jaydeokar**
**What type of PR is this?**
feature

**Which issue does this PR fix**:
N/A

**What does this PR do / Why do we need it**:
To get cluster information during integration tests on deveks cluster

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:
N/A

**Testing done on this change**:
Yes

**Automation added to e2e**:
N/A

**Will this PR introduce any new dependencies?**:
N/A

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
N/A

**Does this change require updates to the CNI daemonset config files to work?**:
N/A

**Does this PR introduce any user-facing change?**:
N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/run-cni-release-tests.sh [hunk 1]
```diff
@@ -29,8 +30,8 @@ function run_integration_test() {
   echo "cni test took $((SECONDS - START)) seconds."
   echo "Running ipamd integration tests"
   START=$SECONDS
-  # NOTE: skipping ipamd_event_test.go until it can be triaged further
-  cd $INTEGRATION_TEST_DIR/ipamd && CGO_ENABLED=0 ginkgo $EXTRA_GINKGO_FLAGS --skip-file=ipamd_event_test.go -v -timeout 90m --no-color --fail-on-pending -- --cluster-kubeconfig="$KUBE_CONFIG_PATH" --cluster-name="$CLUSTER_NAME" --aws-region="$REGION" --aws-vpc-id="$VPC_ID" --ng-name-label-key="$NG_LABEL_KEY" --ng-name-label-val="$NG_LABEL_VAL" || TEST_RESULT=fail
+
+  cd $INTEGRATION_TEST_DIR/ipamd && CGO_ENABLED=0 ginkgo $EXTRA_GINKGO_FLAGS -v -timeout 90m --no-color --fail-on-pending -- --cluster-kubeconfig="$KUBE_CONFIG_PATH" --cluster-name="$CLUSTER_NAME" --aws-region="$REGION" --aws-vpc-id="$VPC_ID" --ng-name-label-key="$NG_LABEL_KEY" --ng-name-label-val="$NG_LABEL_VAL" || TEST_RESULT=fail
```
Comment (jdn5126) [inline scripts/run-cni-release-tests.sh:34]: Why is this being removed? I think `ipamd_event_test` is still flaky

Comment (jaydeokar) [reply scripts/run-cni-release-tests.sh:34]: Oh I thought we fixed it.. I can revert this

File: test/framework/resources/k8s/resources/deployment.go [hunk 2]
```diff
@@ -60,6 +60,11 @@ func (d *defaultDeploymentManager) CreateAndWaitTillDeploymentIsReady(deployment
 func (d *defaultDeploymentManager) DeleteAndWaitTillDeploymentIsDeleted(deployment *v1.Deployment) error {
 	ctx := context.Background()
 	err := d.k8sClient.Delete(ctx, deployment)
+
+	if errors.IsNotFound(err) {
```
Comment (jdn5126) [inline test/framework/resources/k8s/resources/deployment.go:64]: why this change? This looks like it will skip the normal return below

Comment (jaydeokar) [reply test/framework/resources/k8s/resources/deployment.go:64]: So I was trying to make sure we end up cleaning all the resources we create during a test and not stop the cleanup just because of resource doesn't exist error. 
Say if a deployment doesn't exist ( eg: we got an error before the deployment was even created), we shouldn't error out during the cleanup process. 
Specifically, in the After each step, we check the status of delete deployment call and if it returns an not found error, the rest of the cleanup get's missed. (eg - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/test/integration/cni/pod_traffic_test.go#L150)

Comment (jdn5126) [reply test/framework/resources/k8s/resources/deployment.go:64]: I see. So I think the function should still return an error, but we should do all of the cleanup regardless of whether an earlier step returns an error, similar to what we do here: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/test/integration/custom-networking/custom_networking_suite_test.go#L175

Basically, we append errors and check them at the end

File: scripts/lib/cluster.sh [hunk 3]
```diff
@@ -7,6 +7,14 @@ function load_cluster_details() {
   K8S_VERSION=$(echo "$DESCRIBE_CLUSTER_OP" | jq .cluster.version -r)
 }
 
+function load_deveks_cluster_details() {
+
+  echo "loading cluster details $CLUSTER_NAME"
+  PROVIDER_ID=$(kubectl get nodes --kubeconfig $KUBE_CONFIG_PATH -ojson | jq -r '.items[0].spec.providerID')
+  INSTANCE_ID=${PROVIDER_ID##*/}
```
Comment (orsenthil) [inline scripts/lib/cluster.sh:14]: Why do we have `##*/` here? Perhaps for a comment?

Comment (jaydeokar) [reply scripts/lib/cluster.sh:14]: It's a way to fetch the instance id from the string provider id. It removes all the characters upto the last '/'
aws:///us-west-2a/i-xxxxxxxxxxxxx -> i-xxxxxxxxxxxxx

**Review by jdn5126 — APPROVED** [review]

**Review by orsenthil — COMMENTED** [review]
LGTM
